src/internal_logging.h

```diff
@@ -109,8 +109,9 @@ extern PERFTOOLS_DLL_DECL void (*log_message_writer)(const char* msg, int length
 #undef CHECK_CONDITION
 #define CHECK_CONDITION(cond)                                            \
 do {                                                                     \
-  if (!(cond)) {                                                         \
+  if (PREDICT_FALSE(!(cond))) {                                          \
     ::tcmalloc::Log(::tcmalloc::kCrash, __FILE__, __LINE__, #cond);      \
+    UNREACHABLE;                                                         \
```

Comment:
Next time please consider separating this out to distinct commit. This change is unrelated to safe linking. In general I'm slightly opposing this specific change. Not sure about you but those  noreturn/"fatal" calls which crash program end up causing difficulties debugging crash. Because modern compilers tend to conflate call sites in those cases. Your change is slightly different and might work better. But in best world we'd be discussing (and perhaps testing) that separately.

BTW did you do that because it makes specific improvement in generate code or out of general desire to improve ?